src/components/AppController/AppController.js

```diff
@@ -13,7 +13,7 @@ import React from 'react';
 import createReactClass from 'create-react-class';
 import { Grid, Row, Col } from 'react-bootstrap';
 
-import MapController from '../MapController';
+import SingleMapController from '../MapController/SingleMapController';
```

Comment:
Because variants of different controller components are proliferating, and somewhat confusingly named, I'd like to introduce some grouping and naming conventions. Instead of putting specific suggestions all over this PR, I'll say them just once here, with application in many places.:

1. Put all app controllers in a subdirectory `./app-controllers`.
2. Put all data controllers in a subdirectory `./data-controllers`
3. Rename the subdirectory `./MapController` to `./map-controllers`.
2. Use consistent naming for data controller and map controller components. Name pattern is `<Portal><Thing>Controller`, where `<Portal> = Single | Dual | Precip` and `<Thing> = App | Map | Data`.
5. Where applicable (in most if not all of the app and data controllers), do the following:
   1. Just do `export default createReactClass({ ...`. No need to create an intervening variable (often misnamed, e.g.,  `App`) that is immediately exported.
   2. Set `displayName` to the component's name (according to convention above). There are several places where `displayName` is wrong and will show the wrong name in the React debug extensions for Chrome/Firefox.
6. Similarly for map controllers, use an accurate class name, e.g., `export default class DualMapController ...` (not `... MapController ...`).

I know the existing situation is largely legacy, but it's starting to get confusing and ever more inconsistent and this is the time to fix it.

I experimented on a separate branch with renaming, etc., and github is smart enough to follow the renames logically. Yay. Otherwise this would have to go in a separate PR. Just in case something does screw up and the renaming blows up on us, please do all this renaming in separate commit(s) so that they can be backed out easily, and can be cherry-picked over to a separate branch if necessary.

Reply:
@corviday , I would be happy to do this task for you. My IDE has refactoring tools that do 99% of the bookkeeping work associated with this (modifying all the references to the renamed files, etc.), which I know is a real pain to do manually.